program/tests/harvest_rewards.rs

```diff
@@ -0,0 +1,557 @@
+    {
+        context.set_account(
+            &token_account,
+            &AccountSharedData::new_data(100_000_000, &vec![5; 165], &spl_token_2022::id())
+                .unwrap(),
+        );
+    }
+
+    let instruction = harvest_rewards(&holder_rewards_pool, &holder_rewards, &token_account, &mint);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(0, InstructionError::InvalidAccountData)
+    );
+}
+
+#[tokio::test]
+async fn fail_token_account_mint_mismatch() {
+    let owner = Pubkey::new_unique();
+    let mint = Pubkey::new_unique();
+
+    let token_account = get_associated_token_address(&owner, &mint);
+    let holder_rewards = get_holder_rewards_address(&token_account);
+    let holder_rewards_pool = get_holder_rewards_pool_address(&mint);
+
+    let mut context = setup().start_with_context().await;
+    setup_token_account(
+        &mut context,
+        &token_account,
+        &owner,
+        &Pubkey::new_unique(), // Incorrect mint.
+        0,
+    )
+    .await;
+    setup_mint(&mut context, &mint, &Pubkey::new_unique(), 0).await;
+
+    let instruction = harvest_rewards(&holder_rewards_pool, &holder_rewards, &token_account, &mint);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(
+            0,
+            InstructionError::Custom(PaladinRewardsError::TokenAccountMintMismatch as u32)
+        )
+    );
+}
+
+#[tokio::test]
+async fn fail_holder_rewards_pool_incorrect_owner() {
+    let owner = Pubkey::new_unique();
+    let mint = Pubkey::new_unique();
+
+    let token_account = get_associated_token_address(&owner, &mint);
+    let holder_rewards = get_holder_rewards_address(&token_account);
+    let holder_rewards_pool = get_holder_rewards_pool_address(&mint);
+
+    let mut context = setup().start_with_context().await;
+    setup_holder_rewards_account(&mut context, &holder_rewards, 0, 0).await;
+    setup_token_account(&mut context, &token_account, &owner, &mint, 0).await;
+    setup_mint(&mut context, &mint, &Pubkey::new_unique(), 0).await;
+
+    // Setup holder rewards pool account with incorrect owner.
+    {
+        context.set_account(
+            &holder_rewards_pool,
+            &AccountSharedData::new_data(100_000_000, &vec![5; 165], &system_program::id())
```

Comment:
nit: can you make this the right size of 8?

Reply:
Bah, I always forget this step.